### ISSUE TITLE
gradle-plugin: `user.home` for github actions is still incorrect

### ISSUE THREAD

**Issue by petertrr** [saveourtool/diktat#1269]
https://github.com/analysis-dev/diktat/blob/755b6b7421d1a070ba598b48329d0f7468cce2cf/diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt#L195

Should use `rootDir`, not `projectDir`. Current version will yield correct results only for the root project.

**Comment by petertrr** [comment]
Another problem with it:
```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/D:/STUFF/.gradle/caches/modules-2/files-2.1/org.apache.logging.log4j/log4j-slf4j-impl/2.17.2/183f7c95fc981f3e97d008b363341343508848e/log4j-slf4j-impl-2.17.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/D:/STUFF/.gradle/caches/modules-2/files-2.1/ch.qos.logback/logback-classic/1.2.9/7d495522b08a9a66084bf417e70eedf95ef706bc/logback-classic-1.2.9.jar!/org/slf4j/impl/StaticLoggerBinder.class]
```

**Comment by petertrr** [comment]
Also, deprecated proeprties of diktat extension in gradle-plugin can be removed already

**Comment by 0x6675636b796f75676974687562** [comment]
The issue still affects DiKTat 1.2.1, at least on Windows. Minimal project file (using the Groovy DSL):

```groovy
buildscript {
    repositories {
        mavenCentral()
    }

    dependencies {
        classpath group: 'org.jetbrains.kotlin', name: 'kotlin-gradle-plugin', version: kotlin_version
        classpath group: 'org.cqfn.diktat', name: 'diktat-gradle-plugin', version: diktat_version
    }
}

subprojects {
    apply plugin: 'kotlin'
    apply plugin: 'org.cqfn.diktat.diktat-gradle-plugin'

    repositories {
        mavenCentral()
    }

    dependencies {
        implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
    }

    compileKotlin {
        kotlinOptions {
            jvmTarget = "1.8"
        }
    }

    compileTestKotlin {
        kotlinOptions {
            jvmTarget = "1.8"
        }
    }

    diktat {
        inputs {
            it.include("src/**/*.kt", "*.kts", "src/**/*.kts")
            it.exclude("build/**")
        }

        reporter = "sarif"
        debug = false
    }
}
```

This leads to:

```console
D:\test>gradlew diktatCheck -Pdiktat.githubActions=true                                                              
> Task :test-api:diktatCheck FAILED
Exception in thread "main" java.lang.IllegalArgumentException: this and base files have different roots: D:\test\src\main\kotlin\com\example\A.kt and C:\Users\user.
        at kotlin.io.FilesKt__UtilsKt.toRelativeString(Utils.kt:117)
        at kotlin.io.FilesKt__UtilsKt.relativeTo(Utils.kt:128)
        at com.pinterest.ktlint.reporter.sarif.SarifReporter.onLintError(SarifReporter.kt:52)
        at com.pinterest.ktlint.core.Reporter$Companion$from$1.onLintError(Reporter.kt:41)
        at com.pinterest.ktlint.internal.KtlintCommandLine.report(KtlintCommandLine.kt:332)
        at com.pinterest.ktlint.internal.KtlintCommandLine.access$report(KtlintCommandLine.kt:48)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$4.invoke(KtlintCommandLine.kt:283)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$4.invoke(KtlintCommandLine.kt:283)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:578)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:536)
        at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:283)
        at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:235)
        at com.pinterest.ktlint.Main.main(Main.kt:31)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test-api:diktatCheck'.
> Process 'command 'C:\Program Files\Zulu\zulu-8\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed
```

The workaround is to override `user.home`:

```groovy
import org.cqfn.diktat.plugin.gradle.DiktatJavaExecTaskBase

// ...

tasks.withType(DiktatJavaExecTaskBase).configureEach {
    systemProperty("user.home", rootDir.toString())
}
```

**Comment by petertrr** [comment]
Discussed; decided to enable workaround for `user.home` not only with `githubActions == true`, but with `reporter == "sarif"`
